mealplanner-ui/src/pages/Meals/Meal.tsx

```diff
@@ -322,51 +310,44 @@ export const Meal = () => {
                     </tr>
                   </thead>
                   <tbody>
-                    {arrayOfIngredientsWithoutSubstituteIngredients?.map((ingredient) => {
+                    {allIngredients?.map((ingredient) => {
                       return (
-                        <tr key={ingredient.id}>
+                        <tr key={ingredient.rowId}>
                           <td>
-                            {ingredient.name}
-                            {ingredient.substituteIngredient ? (
+                            {ingredient.substituteIngredientId === null && ingredient.name}
+                            {ingredient.substituteIngredientId && (
                               <>
-                                <br />
                                 <span style={{ fontStyle: "italic", marginLeft: "0.5rem" }}>
-                                  Substitute: {ingredient.substituteIngredient.name}
+                                  Substitute: {ingredient?.name}
                                 </span>
                                 <br />
                                 <span style={{ fontStyle: "italic", marginLeft: "0.5rem" }}>
-                                  Reason:{" "}
+                                  Reason:
                                   {ingredient.substituteReason
                                     ? ingredient.substituteReason
                                     : "Not specified"}
                                 </span>
                               </>
-                            ) : (
-                              <></>
                             )}
                           </td>
                           <td style={{ textAlign: "center", verticalAlign: "top" }}>
-                            {ingredient.quantity}
-                            {ingredient.substituteIngredient ? (
+                            {ingredient.substituteIngredientId === null && ingredient.quantity}
+                            {/* {ingredient.quantity} */}
+                            {ingredient.substituteIngredient && (
                               <>
-                                <br />
-                                <span>{ingredient.substituteIngredient.quantity}</span>
+                                <span>{ingredient.quantity}</span>
                                 <br />
                               </>
-                            ) : (
-                              <></>
                             )}
                           </td>
                           <td style={{ paddingLeft: "0.5rem" }}>
-                            {ingredient.unit}
-                            {ingredient.substituteIngredient ? (
+                            {ingredient.substituteIngredientId === null && ingredient.unit}
```

Comment:
Why do we need this check for ingredient.unit. Unit exists irrespective of whether it is substitute Ingredient isnt it?